pkg/config/config.go

```diff
@@ -281,16 +282,18 @@ func (c *MicroshiftConfig) ReadAndValidate(flags *pflag.FlagSet) error {
 	return nil
 }
 
-func InitGlobalFlags() {
-	pflag.CommandLine.SetNormalizeFunc(cliflag.WordSepNormalizeFunc)
+func HideUnsupportedFlags(flags *pflag.FlagSet) {
+	// hide logging flags that we do not use/support
+	loggingFlags := pflag.NewFlagSet("logging-flags", pflag.ContinueOnError)
+	logs.AddFlags(loggingFlags)
+
+	supportedLoggingFlags := sets.NewString("v")
 
-	goflag.CommandLine.VisitAll(func(goflag *goflag.Flag) {
-		if StringInList(goflag.Name, []string{"v", "log_file"}) {
-			pflag.CommandLine.AddGoFlag(goflag)
+	loggingFlags.VisitAll(func(pf *pflag.Flag) {
+		if !supportedLoggingFlags.Has(pf.Name) {
+			flags.MarkHidden(pf.Name)
 		}
 	})
 
-	pflag.CommandLine.MarkHidden("log-flush-frequency")
-	pflag.CommandLine.MarkHidden("log_file")
-	pflag.CommandLine.MarkHidden("version")
+	flags.MarkHidden("version")
```

Comment:
I am not sure what flags does microshift want to expose and which not. Most of them are deprecated except `-v`, `--vmodule` and `log-flush-frequency`.

This is the output with `HideUnsupportedFlags` on

```
MicroShift, a minimal OpenShift

Usage:
  microshift [flags]
  microshift [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  run         Run MicroShift
  show-config Print MicroShift's configuration
  version     Print MicroShift version information

Flags:
  -h, --help      help for microshift
  -v, --v Level   number for the log level verbosity

Use "microshift [command] --help" for more information about a command.
```

and this without

```
MicroShift, a minimal OpenShift

Usage:
  microshift [flags]
  microshift [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  run         Run MicroShift
  show-config Print MicroShift's configuration
  version     Print MicroShift version information

Flags:
      --add-dir-header                   If true, adds the file directory to the header of the log messages (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --alsologtostderr                  log to standard error as well as files (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
  -h, --help                             help for microshift
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-dir string                   If non-empty, write log files in this directory (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-file string                  If non-empty, use this log file (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-flush-frequency duration     Maximum number of seconds between log flushes (default 5s)
      --logtostderr                      log to standard error instead of files (default true) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --one-output                       If true, only write logs to their native severity level (vs also writing to each lower severity level) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --skip-headers                     If true, avoid header prefixes in the log messages (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --skip-log-headers                 If true, avoid headers when opening log files (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
  -v, --v Level                          number for the log level verbosity
      --version version[=true]           Print version information and quit
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging


```

Reply:
I'm ok with only leaving the `-v` flag. We want MicroShift to log to the journal where users would have log rotation/retention policies. I don't see a reason to introduce deprecated flags at this point, either.